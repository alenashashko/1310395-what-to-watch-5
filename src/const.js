export const MillisecondsInTimePeriod = {
  SECOND: 1000,
  MINUTE: 60000,
  HOUR: 3600000,
};
