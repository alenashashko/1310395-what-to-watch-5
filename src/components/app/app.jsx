import React from 'react';
import {Switch, Route, Router} from 'react-router-dom';
import {PropTypes} from 'prop-types';
import {connect} from 'react-redux';

import proptypes from '../../type';
import MainPage from '../main-page/main-page';
import AuthPage from '../auth-page/auth-page';
import MyListPage from '../my-list-page/my-list-page';
import MoviePage from '../movie-page/movie-page';
import AddReviewPage from '../add-review-page/add-review-page';
import PlayerPageWrapped from '../player-page/player-page';
import PrivateRoute from '../private-route/private-route';
import {AuthorizationStatus, AppRoute} from '../../const';
import {getAuthorizationStatus} from '../../store/selectors';
import browserHistory from '../../browser-history';

const App = (props) => {
  const {cinemaName, authorizationStatus} = props;

  if (authorizationStatus === AuthorizationStatus.UNKNOWN) {
    return null;
  }

  return (
    <Router history={browserHistory} basename={AppRoute.ROOT}>
      <Switch>
        <Route exact path={AppRoute.ROOT} render={() => {
          return (
            <MainPage
              cinemaName={cinemaName}
            />
          );
        }}>
        </Route>
        <Route exact path={AppRoute.LOGIN}>
          <AuthPage cinemaName={cinemaName} />
        </Route>
        <PrivateRoute
          exact
          path={AppRoute.MY_LIST}
          render={() => (
            <MyListPage cinemaName={cinemaName} />
          )}
        >
        </PrivateRoute>
        <Route exact path={AppRoute.MOVIE} render={(routeProps) => {
          const {id} = routeProps.match.params;

          return (
            <MoviePage
              id={id}
              key={id}
              cinemaName={cinemaName}
            />
          );
        }}>
        </Route>
        <PrivateRoute
          exact
          path={AppRoute.REVIEW}
          render={(routeProps) => {
            return (
              <AddReviewPage
                cinemaName={cinemaName}
                id={routeProps.match.params.id}
              />
            );
          }}>
        </PrivateRoute>
        <Route exact path={AppRoute.PLAYER} render={(routeProps) => (
          <PlayerPageWrapped id={routeProps.match.params.id} />
        )}>
        </Route>
      </Switch>
    </Router>
  );
};

App.propTypes = {
  cinemaName: proptypes.cinemaName,
  authorizationStatus: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state)
});

export {App};
export default connect(mapStateToProps)(React.memo(App));
