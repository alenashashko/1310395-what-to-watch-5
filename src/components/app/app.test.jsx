import React from 'react';
import renderer from 'react-test-renderer';

import {App} from './app';
import {AuthorizationStatus} from '../../const';

jest.mock(`../main-page/main-page`, () => {
  const MainPageMock = () => <p>MainPageMock</p>;

  return {
    __esModule: true,
    default: MainPageMock
  };
});

describe(`Should App render correctly`, () => {
  it(`With AUTH authorization status`, () => {
    const tree = renderer
      .create(
          <App
            authorizationStatus={AuthorizationStatus.AUTH}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With NO_AUTH authorization status`, () => {
    const tree = renderer
      .create(
          <App
            authorizationStatus={AuthorizationStatus.NO_AUTH}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With UNKNOWN authorization status`, () => {
    const tree = renderer
      .create(
          <App
            authorizationStatus={AuthorizationStatus.UNKNOWN}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
