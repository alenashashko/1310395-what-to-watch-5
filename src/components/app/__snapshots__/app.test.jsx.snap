// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly With AUTH authorization status 1`] = `
<p>
  MainPageMock
</p>
`;

exports[`Should App render correctly With NO_AUTH authorization status 1`] = `
<p>
  MainPageMock
</p>
`;

exports[`Should App render correctly With UNKNOWN authorization status 1`] = `null`;
